phantomjs-env/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala

```diff
@@ -119,37 +92,50 @@ class PhantomJSEnv(config: PhantomJSEnv.Config)
       val ctors = clazz.getConstructors()
       assert(ctors.length == 1, "JettyWebsocketManager may only have one ctor")
 
-      val mgr = ctors.head.newInstance(websocketListener)
+      val listener = new WebsocketListener {
+        def onRunning(): Unit = startPhantomJSProcess()
+        def onOpen(): Unit = sendPendingMessages()
```

Comment:
Please abstract this over Futures so there is less callback. It might be a good idea to adapt our shim for classloading to give access to futures (and only take an `onMessage` and `log` handle).